### PR TITLE
Load reference name in LoadPBI()

### PR DESCRIPTION
**Pull request by tian52**

### CODE REVIEW

File: src/LoadData.cpp [hunk 1]
```diff
@@ -361,14 +361,15 @@ DataFrame loadPBI(std::string filename,
     auto mappedData = raw.MappedData();
     IntegerVector mappedtId(wrap(mappedData.tId_));
     mappedtId = mappedtId + 1;
-    try {
-      BamReader br(filename);
+    for(auto fn : fileNames) {
```
Comment (evolvedmicrobe) [inline src/LoadData.cpp]: This is a definite improvement! On thing about this though, is that you are resetting the sequence name levels for every filename in the BAM.  A better approach might be the following:

1 - Load the first BAM file (if it exists) and set the header levels equal to the sequences in that file.
2 - For the remaining BAMs, check that the sequences in their headers are the same as in the first BAM (so sequence 1 in the second file is the same as the first file, etc.).  If not, we can `stop(` with an error to indicate that this isn't the case (in practice this error should basically never occur so we don't need to worry to much about handling it, we can just catch it).

File: tests/testthat/test.load.R [hunk 2]
```diff
@@ -32,6 +32,11 @@ test_that("loadFromDataFrame", {
   expect_equal(length(aln), 3)
 })
 
+test_that("loadRefName", {
+  d = loadPBI(bfile)
```
Comment (evolvedmicrobe) [inline tests/testthat/test.load.R]: This file is a BAM file so this test would have passed even before this change was made, I would make the test use the `bamdset` variable which is equal to the alignmentset file `"AlignmentSet/m54006_160504_020705.alignmentset.xml"`

**Comment by dalexander** [comment]
Thanks for addressing Nigel's request and adding the test.  A couple comments.

In the future, please make the commit messages abundantly clear about what behavior is being changed.  As best I can tell here, the essential change is that reference names are now correctly looked up for _dataset_ inputs, whereas before it only worked for BAM files.  That kind of detail is crucial for commit messages.  You should use multiple lines in your commit message---the first line should be a short summary, then a blank line, and then some more details as needed.

Your fix to the looping logic should have been in a separate commit from the unit test, each commit with an appropriate commit message.

The indentation gets weird at line 368, I don't know why.  Can you try to fix that?

Finally, have you verified that things work as expected with multiple references sequences?

**Comment by tian52** [comment]
@dalexander Thanks a lot for the comments! I have corrected the indentation and made a new commit with a more clear message. Please let me know if it still needs to be improved.
I have verified the function for multiple references locally. The figure below shows the results of running `cdtest = loadPBI("/pbi/dept/secondary/siv/smrtlink/smrtlink-internal/userdata/jobs-root/005/005573/tasks/pbalign.tasks.consolidate_alignments-0/combined.alignmentset.xml")`. Should I also add a unit test to test it works as expected for multiple references?
screen shot 2017-01-17 at 12 14 42 pm

**Comment by dalexander** [comment]
It looks like there's probably no multi-reference-contig dataset available in pbbamr, and I don't want to give you too much trouble for this change.  So let's not worry about adding another unit test ATM, I am satisfied with the result you have shown here.

**Comment by tian52** [comment]
Thanks for approving it!

**Pull request merged**